examples/tests/sdk_tests/bucket/metadata.test.w

```diff
@@ -0,0 +1,17 @@
+bring cloud;
+
+let b = new cloud.Bucket();
+
+test "metadata" {
+  b.put("test1.txt", "Foo");
+
+  assert(b.metadata("test1.txt").size == 3);
+  assert(b.metadata("test1.txt").contentType == nil);
```

Comment:
When I ran this test on my AWS account, I got an error because the contentType was not nil. To make sure the AWS and `sim` targets behave the same, I think we need the following:

- (a) `put` for AWS implementations should always specify a content type so it isn't automatically set by AWS - I think the AWS SDK already sets it to [application/octet-stream](https://github.com/aws/aws-sdk-js-v3/blob/fa961c2350b650a1c29cc09306f0e2e0b9eb6c4e/clients/client-s3/src/protocols/Aws_restXml.ts#L3320) by default but it's a good practice for us to set it
- (b) the `sim` implementation should track a content type for each file, such as inside a JavaScript `Map`

Reply:
Ah, I thought this test is executed only in Simulator, but we can execute in actual clouds?
How should I execute this test file in AWS?
I've checked `05-development.md` but could not figure out.

Reply:
In my personal opinion, there should be a `content-type` argument to `put` methods and let users to specify it.
It would be user-friendly if content-type is automatically guessed by Wing using a 3rd party library, but there should be an escape hatch in case such a library does not recognize content-type.

Reply:
Sorry for any confusion with our development guide. You can execute any test in AWS by running Wing's  `test` command with the `--target tf-aws` flag. For example `wing test --target tf-aws metadata.test.w`.

> In my personal opinion, there should be a `content-type` argument to `put` methods and let users to specify it. It would be user-friendly if Wing automatically guessed content-type using a 3rd party library, but there should be an escape hatch in case such a library does not recognize content-type.

Totally, I think both suggestions make sense to me. I didn't want to add additional scope to this PR, but both can be supported. Would you mind creating an issue on GitHub for these enhancements? 🙏

Reply:
@Chriscbr 
Updated the PR. 
- https://github.com/winglang/wing/pull/4338/commits/350470283cb4344e4890eb50ef5d80607975465f Sim returns `application/octet-stream` until we allow custom content-type.
- https://github.com/winglang/wing/pull/4338/commits/34456836cdd784c006948586672b6847794ecbb5 Fixes AWS implementation. I've found that a generic `"403"` error is thrown instead of `NotFound` when `s3:ListBucket` (underlying action for `s3:ListObject(V2)`) is not permitted.

As far as I tested, the test now succeeds both on Simulator and TF-AWS.